src/libraries/System.Memory/tests/Span/IndexOfAnyInRange.cs

```diff
@@ -0,0 +1,200 @@
+
+            AssertExtensions.Throws<ArgumentNullException>("highInclusive", () => IndexOfAnyInRange(Array.Empty<RefType>(), new RefType { Value = 42 }, null));
+            AssertExtensions.Throws<ArgumentNullException>("highInclusive", () => LastIndexOfAnyInRange(Array.Empty<RefType>(), new RefType { Value = 42 }, null));
+            AssertExtensions.Throws<ArgumentNullException>("highInclusive", () => IndexOfAnyExceptInRange(Array.Empty<RefType>(), new RefType { Value = 42 }, null));
+            AssertExtensions.Throws<ArgumentNullException>("highInclusive", () => LastIndexOfAnyExceptInRange(Array.Empty<RefType>(), new RefType { Value = 42 }, null));
+        }
+
+        [Fact]
+        public void NullPermittedInInput()
+        {
+            RefType[] array = new[] { null, new RefType { Value = 1 }, null, null, new RefType { Value = 42 }, null };
+            RefType low = new RefType { Value = 41 }, high = new RefType { Value = 43 };
+
+            Assert.Equal(4, IndexOfAnyInRange(array, low, high));
+            Assert.Equal(4, LastIndexOfAnyInRange(array, low, high));
+
+            Assert.Equal(0, IndexOfAnyExceptInRange(array, low, high));
+            Assert.Equal(5, LastIndexOfAnyExceptInRange(array, low, high));
+        }
+    }
+
+    public class RefType : IComparable<RefType>
+    {
+        public int Value { get; set; }
+        public int CompareTo(RefType? other) => other is null ? 1 : Value.CompareTo(other.Value);
+    }
+
+    public abstract class IndexOfAnyInRangeTests<T>
+        where T : IComparable<T>
+    {
+        protected abstract T Create(int value);
+
+        [Fact]
+        public void EmptyInput_ReturnsNegative1()
+        {
+            T[] array = Array.Empty<T>();
+            T low = Create(0);
+            T high = Create(255);
+
+            Assert.Equal(-1, IndexOfAnyInRange(array, low, high));
+            Assert.Equal(-1, LastIndexOfAnyInRange(array, low, high));
+            Assert.Equal(-1, IndexOfAnyExceptInRange(array, low, high));
+            Assert.Equal(-1, LastIndexOfAnyExceptInRange(array, low, high));
+        }
+
+        [Fact]
+        public void NotFound_ReturnsNegative1()
+        {
+            T fillValue = Create(42);
+            for (int length = 0; length < 128; length++)
+            {
+                T[] array = Enumerable.Repeat(fillValue, length).ToArray();
+
+                Assert.Equal(-1, IndexOfAnyInRange(array, Create(43), Create(44)));
+                Assert.Equal(-1, LastIndexOfAnyInRange(array, Create(43), Create(44)));
+                Assert.Equal(-1, IndexOfAnyExceptInRange(array, Create(42), Create(44)));
+                Assert.Equal(-1, LastIndexOfAnyExceptInRange(array, Create(42), Create(44)));
+            }
+        }
+
+        [Fact]
+        public void FindValueInSequence_ReturnsFoundIndexOrNegative1()
+        {
+            foreach (int length in Enumerable.Range(1, 64))
```

Comment:
I hope nobody will say "I've learned a new perf trick from Stephen Toub, let's replace all our for loops with it" ;)